src/hooks/useConnectionStatus.ts

```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from "react"
+import NetInfo from "@react-native-community/netinfo"
+
+export const useConnectionStatus = (): boolean | null | undefined => {
+  const [isConnected, setIsConnected] = useState<boolean | null | undefined>(
```

Comment:
How do you feel about narrowing down the type, since we are setting up this value, we could always set a boolean and avoid the union type right?